src/Interceptor/Hsts/CombinationHstsJar.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Amp\Http\Client\Interceptor\Hsts;
+
+class CombinationHstsJar implements HstsJar
```

Comment:
Not sure if this would be a helpful thing to include - this would allow you to do something like `CombinationHstsJar(InMemoryHstsJar, GooglePreloadHstsJar)` to mimic standard browser behavior - anything on the preload list already would be promoted before you accessed it, but any new sites you access which advertise themselves as HSTS would be added to the jar.